e2e/nstemplateset_test.go

```diff
@@ -0,0 +1,140 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/doubles"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type nsTemplateSetTest struct {
+	suite.Suite
+	namespace   string
+	testCtx     *framework.TestCtx
+	awaitility  *wait.Awaitility
+	memberAwait *wait.MemberAwaitility
+}
+
+func TestNSTemplateSet(t *testing.T) {
+	suite.Run(t, &nsTemplateSetTest{})
+}
+
+func (s *nsTemplateSetTest) SetupSuite() {
+	nsTmplSetList := &toolchainv1alpha1.NSTemplateSetList{}
+	s.testCtx, s.awaitility = doubles.InitializeOperators(s.T(), nsTmplSetList)
+	s.memberAwait = s.awaitility.Member()
+	s.namespace = s.awaitility.MemberNs
+}
+
+func (s *nsTemplateSetTest) TestCreateOK() {
+	t := s.T()
+	username := "amar"
+
+	// create NSTmplSet
+	t.Logf("Creating NSTmplSet with username:%s", username)
+	nsTmplSet := newNSTmplSet(username, s.namespace)
+	err := s.awaitility.Client.Create(context.TODO(), nsTmplSet, doubles.CleanupOptions(s.testCtx))
+	require.NoError(t, err)
+
+	// wait for NSTmplSet
+	err = s.memberAwait.WaitForNSTmplSet(nsTmplSet.Name)
+	require.NoError(t, err)
+
+	// wait for Namespace
+	for _, ns := range nsTmplSet.Spec.Namespaces {
+		err = s.memberAwait.WaitForNamespace(username, ns.Type)
+		require.NoError(t, err)
+	}
+}
+
+func (s *nsTemplateSetTest) TestDeleteNamespaceOK() {
+	t := s.T()
+	username := "amit"
+
+	// create NSTmplSet
+	t.Logf("Creating NSTmplSet with username:%s", username)
+	nsTmplSet := newNSTmplSet(username, s.namespace)
+	err := s.awaitility.Client.Create(context.TODO(), nsTmplSet, doubles.CleanupOptions(s.testCtx))
+	require.NoError(t, err)
+
+	// wait for NSTmplSet
+	err = s.memberAwait.WaitForNSTmplSet(nsTmplSet.Name)
+	require.NoError(t, err)
+
+	// wait for Namespace
+	for _, ns := range nsTmplSet.Spec.Namespaces {
+		err = s.memberAwait.WaitForNamespace(username, ns.Type)
+		require.NoError(t, err)
+	}
```

Comment:
the lines 59-74 are same as in the previous test `TestCreateOK` - to avoid code duplication and improve readability, we can create a subtest inside of `TestCreateOK` that will contain code from lines 76-85

Reply:
Done https://github.com/codeready-toolchain/toolchain-e2e/pull/5/commits/4e5ce0b205d225343ba4508aba51b8c7aa16fe82